org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java

```diff
@@ -267,7 +284,7 @@ public static class LanguageServerDocumentExecutor extends LanguageServers<Langu
 		@Override
 		public boolean anyMatching() {
 			return LanguageServiceAccessor.getLSWrappers(document).stream()
-					.map(this::filter).map(CompletableFuture::join).anyMatch(Objects::nonNull);
+					.map(this::filter).map(this::matches).anyMatch(bool -> bool);
```

Comment:
```suggestion
			return !LanguageServiceAccessor.getLSWrappers(document).stream()
					.map(this::filter).map(this::matches).isEmpty();
```

Reply:
thanks, this kind of test makes more sense. There is no isEmpty on an stream but I have used now `.count() != 0`

Reply:
Sorry, it's just `.empty()`. Please avoid using `.count()` as it will evaluate all content of the stream, the complexity is `O(n)` instead of `O(1)`

Reply:
My IDE tells me there is no `empty()` either. Am I doing something wrong? 

So far I have used .findFirst().isPresent() in https://github.com/eclipse/lsp4e/pull/467

Reply:
My bad again, `empty()` is an empty constructor; .findFirst().isPresent() is good.